BCDA-2302 Accessibility: Visible focus state

dhgreene opened the pull request:
### Fixes [BCDA-2302](https://jira.cms.gov/browse/BCDA-2302)
The CMS Design System elements until recently had subtle, color-only indications of focus.  The goal of this PR is to take advantage of recent updates to the design system that make focus more obvious.

### Proposed Changes
- Isolate any CSS changes BCDA had made outside of the commented BCDA-specific section
  - Original CSS lines 54-55, 79-80, 7344
- Update CMS Design System Core CSS
  - Original CSS lines 243-4997

Note that the CMS Design System Layout has not been altered (lines 4997-6041).

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

This is a format-only change.

### Acceptance Validation
#### Examples of the new focus indicators at work
![Screen Shot 2020-03-04 at 3 07 57 PM](https://user-images.githubusercontent.com/2533561/75926872-46ab4e00-5e39-11ea-8188-a3e54a3cafea.png)
![Screen Shot 2020-03-04 at 3 07 14 PM](https://user-images.githubusercontent.com/2533561/75926875-4743e480-5e39-11ea-97d6-4bcc0fb20004.png)
![Screen Shot 2020-03-04 at 3 06 47 PM](https://user-images.githubusercontent.com/2533561/75926877-47dc7b00-5e39-11ea-9018-8df76131e787.png)
![Screen Shot 2020-03-10 at 1 35 38 AM](https://user-images.githubusercontent.com/2533561/76282875-80cc8380-626f-11ea-82a6-047ed4216cd2.png)
![Screen Shot 2020-03-10 at 1 35 20 AM](https://user-images.githubusercontent.com/2533561/76282877-81651a00-626f-11ea-9083-b196f2f6dec5.png)
![Screen Shot 2020-03-10 at 1 35 08 AM](https://user-images.githubusercontent.com/2533561/76282878-81651a00-626f-11ea-938f-fcc91d30c313.png)



### Feedback Requested
- Please run this locally, as there is a potential impact on unexpected design elements.  Does all look OK?
- Based on the screenshots/local experience, do we need to make any custom changes to visible focus state?  In particular, is the color (new! custom!) for selected links appropriate, or should we choose a different one?

msnook commented:
Could you sync this feature branch with `master` so I know I'm pulling the latest when I test locally?

dhgreene commented:
> Could you sync this feature branch with `master` so I know I'm pulling the latest when I test locally?

@msnook Yes, ready now.

msnook commented on assets/css/static-main.css:
```diff
@@ -466,13 +546,89 @@ a:active {
+}
+
+.ds-c-link:focus, .ds-c-link:focus:visited, a:focus, a:focus:visited {
+    background-color: #ffde11;
```
Is this a color that is required by 508 Compliance, or one that we selected?

dhgreene replied:
@msnook It's the color chosen by the design system.  I've overridden it in the last commit; do you like this better?

dhgreene commented:
> (1) Links, once clicked, turn a different color (a color that I would consider to be outside the scope of the visible focus state). For example, in screenshot, I clicked on `home` and now it remains blue/purple. Is that an un-intended side effect?

@msnook I've fixed this for the top menu.  Having the purple color for visited links elsewhere isn't jarring, IMO.

> (2) Links that open in a different window remain in the visible focus state. For example, clicking on the `BCDA Production Onboarding Request Form` opens a new page; but then navigation back to the original page indicates that the focus is still on `BCDA Production Onboarding Request Form`. Is this an un-intended side effect?

This would be intended, because in the original window that link still has focus.  That look/behavior seems more familiar if you try tabbing between links (using a browser that allows tab navigation to links, e.g. NOT Firefox on OS X unless you've changed settings).

davidakennedy commented:
Hey @dhgreene, thanks for working on this and pushing it forward. Lots of great progress here on what is a big change.

I've retested and have some feedback:

The skip link has some transparency issues when focused. See screenshot.

![Screen Shot 2020-03-09 at 3 54 38 PM](https://user-images.githubusercontent.com/1473618/76256427-4ee2ff00-6226-11ea-93f7-63255e264178.png)

The link to the Google Group in the top right corner doesn't meet color contrast requirements when focused. See screenshot.

![Screen Shot 2020-03-09 at 3 54 59 PM](https://user-images.githubusercontent.com/1473618/76256557-881b6f00-6226-11ea-8e65-3af8a141fc82.png)

The back to top button still has a yellow background. See screenshot.

![Screen Shot 2020-03-09 at 3 57 07 PM](https://user-images.githubusercontent.com/1473618/76256651-ad0fe200-6226-11ea-805e-e335b1923986.png)

The blue background on the horizontal menu doesn't meet minimum color contrast requirements when links are focused. See screenshot.

![Screen Shot 2020-03-09 at 4 46 03 PM](https://user-images.githubusercontent.com/1473618/76257111-6ff81f80-6227-11ea-8b89-49cb08380c27.png)

The accordions need a visible focus style too. We don't have a component for this in the design system, but the button focus styles should work. But if you all feel that doesn't fit, let me know, and I can help with some ideas. See screenshot.

![Screen Shot 2020-03-09 at 4 18 25 PM](https://user-images.githubusercontent.com/1473618/76257199-a2a21800-6227-11ea-9b73-b77dd6099b1b.png)

Lastly, if I could offer some design feedback on the focus color. If you don't want to go with yellow, to help fit the feel of the site better, maybe try a shade of the green from the gradient in the hero area? Maybe this as the background color: `#68C1BF`? That way, it mimics what the yellow does (it light and bright), but isn't yellow.

That would look something like:

![Screen Shot 2020-03-09 at 5 09 47 PM](https://user-images.githubusercontent.com/1473618/76257957-0aa52e00-6229-11ea-8a89-4ccb81428551.png)

Let me know if you have any questions!

dhgreene commented:
@davidakennedy Thanks for catching those details, and for the color suggestion.  The latest commit addresses your review.  Here are some screenshots showing the new behavior:

#### Contrast for links in focus
![Screen Shot 2020-03-10 at 1 22 25 AM](https://user-images.githubusercontent.com/2533561/76282414-f8011800-626d-11ea-8121-88e89fb1ef65.png)
#### "Back to top" button styled properly
![Screen Shot 2020-03-10 at 1 21 42 AM](https://user-images.githubusercontent.com/2533561/76282415-f8011800-626d-11ea-978b-c8b285f2651d.png)
#### "Copy to clipboard" button active
![Screen Shot 2020-03-10 at 1 21 13 AM](https://user-images.githubusercontent.com/2533561/76282416-f8011800-626d-11ea-82ad-d1e3ce70f0cf.png)
#### Accordion active
![Screen Shot 2020-03-10 at 1 20 57 AM](https://user-images.githubusercontent.com/2533561/76282417-f899ae80-626d-11ea-81d4-1de6b27997c4.png)
#### Top menu active with proper contrast
![Screen Shot 2020-03-10 at 1 20 43 AM](https://user-images.githubusercontent.com/2533561/76282418-f899ae80-626d-11ea-9973-7c52516363c6.png)
#### Join the Google Group with proper contrast
![Screen Shot 2020-03-10 at 1 20 20 AM](https://user-images.githubusercontent.com/2533561/76282419-f899ae80-626d-11ea-8b0d-6975a6eca0da.png)
#### Skip to main content: opaque
![Screen Shot 2020-03-10 at 1 20 09 AM](https://user-images.githubusercontent.com/2533561/76282420-f899ae80-626d-11ea-8a1b-9f00f38f5e45.png)

lianilychee commented:
lgtm
